### PR TITLE
Improve Dart examples.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: dart/example/pubsub/stream.dart [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import 'package:toit_api/toit/api/pubsub/subscribe.pbgrpc.dart';
+import 'package:toit_api/toit/model/pubsub/message.pb.dart';
+import 'package:uuid/uuid.dart';
+
+import 'subscription.dart';
+
+const String topic = "cloud:toit-api/example";
+
+/// Whether the given [error] is an error where the application should just retry.
+bool isRetryableGrpcError(GrpcError error) {
```
Comment (floitsch) [inline dart/example/pubsub/stream.dart:20]: You used a different set of retryable errors in the streamsheet code.
Please have a look here, whether mine are correct or not.
(I went through the list of errors and tried to pick the right ones).

File: dart/example/pubsub/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+With the temporary subscription, the listening program only gets the
+data that is published while the subscription is alive. This can be used to
+"listen in" on data that is published.
+
+## Run
+
+Get an API Key from the Toit server:
+``` shell
+  toit org api-keys add <name>
```
Comment (jstroem) [inline dart/example/pubsub/README.md:27]: needs to be  `toit project api-keys ...`

File: dart/example/pubsub/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  toit org api-keys print-secret <name>
```
Comment (jstroem) [inline dart/example/pubsub/README.md:28]: `toit project api-keys print-secret <api-key-id>`

File: dart/example/pubsub/stream.dart [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  switch (error.code) {
+    case StatusCode.ok:
+    case StatusCode.unknown:
+    case StatusCode.deadlineExceeded:
+    case StatusCode.unavailable:
+      return true;
+
+    case StatusCode.cancelled:
```
Comment (jstroem) [inline dart/example/pubsub/stream.dart:28]: retry this

File: dart/example/pubsub/stream.dart [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    case StatusCode.invalidArgument:
+    case StatusCode.notFound:
+    case StatusCode.alreadyExists:
+    case StatusCode.permissionDenied:
+    case StatusCode
+        .resourceExhausted: // Could also be acceptable for retry.case
```
Comment (jstroem) [inline dart/example/pubsub/stream.dart:34]: retry this

File: dart/example/pubsub/stream.dart [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    case StatusCode.failedPrecondition:
+    case StatusCode.aborted:
+    case StatusCode.outOfRange:
+    case StatusCode.unimplemented:
+    case StatusCode.internal:
```
Comment (jstroem) [inline dart/example/pubsub/stream.dart:39]: retry this

File: dart/example/pubsub/stream.dart [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    case StatusCode.dataLoss:
+    case StatusCode.unauthenticated:
+    default:
+      return false;
+  }
```
Comment (jstroem) [inline dart/example/pubsub/stream.dart:44]: IMO you only list the status codes that should be retried otherwise (`default:`) `return false`

**Review by jstroem — APPROVED** [review]

File: dart/example/pubsub/README.md (hunk 2 above)
Comment (floitsch) [reply dart/example/pubsub/README.md:27]: done

File: dart/example/pubsub/README.md (hunk 3 above)
Comment (floitsch) [reply dart/example/pubsub/README.md:28]: done.

File: dart/example/pubsub/stream.dart (hunk 4 above)
Comment (floitsch) [reply dart/example/pubsub/stream.dart:28]: done.

File: dart/example/pubsub/stream.dart (hunk 5 above)
Comment (floitsch) [reply dart/example/pubsub/stream.dart:34]: done.

File: dart/example/pubsub/stream.dart (hunk 6 above)
Comment (floitsch) [reply dart/example/pubsub/stream.dart:39]: done.

File: dart/example/pubsub/stream.dart (hunk 7 above)
Comment (floitsch) [reply dart/example/pubsub/stream.dart:44]: I think it's nicer to show that you thought about these errors.
You could still have forgotten one which falls under the `default`, but at least I have some idea of which errors won't be retried.

**Pull request merged**